### PR TITLE
fix(cli/fetch): fix relative redirect

### LINKED ISSUE #6708: Fetch seems to be fetching wrongly formatted URL when it involves redirect
When trying to fetch a [Wikipedia REST API](https://en.wikipedia.org/api/rest_v1/#/Page%20content/get_page_summary__title_), Deno will thrown error saying failed to connect, while the url displayed in the error message is not a valid URL. With further testing, it seems that the API will have some redirect behind.

---

Sample failure code:
```ts
fetch("https://en.wikipedia.org/api/rest_v1/page/summary/computer").then((res)=> {
  res.json().then((data)=> {
    console.log(data)
  });
});
```

Error:
```
error: Uncaught Http: error sending request for url (https://en.wikipedia.orgcomputer/): error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
```

---

Similar code which the request is without redirect:
```ts
fetch("https://en.wikipedia.org/api/rest_v1/page/summary/Computer").then((res)=> {
  res.json().then((data)=> {
    console.log(data)
  });
});
```

Response: 
```json
{
  "type": "standard",
  "title": "Computer",
  "displaytitle": "Computer",
  "namespace": {
    "id": 0,
    "text": ""
  },
  "wikibase_item": "Q68",
  "titles": {
    "canonical": "Computer",
    "normalized": "Computer",
    "display": "Computer"
  },
  "pageid": 7878457,
  "lang": "en",
  "dir": "ltr",
  "revision": "966740228",
  "tid": "1d4a2fc0-c3cb-11ea-a890-c1085d3174ec",
  "timestamp": "2020-07-08T22:21:37Z",
  "description": "Automatic general-purpose device for performing arithmetic or logical operations",
  "description_source": "local",
  "content_urls": {
    "desktop": {
      "page": "https://en.wikipedia.org/wiki/Computer",
      "revisions": "https://en.wikipedia.org/wiki/Computer?action=history",
      "edit": "https://en.wikipedia.org/wiki/Computer?action=edit",
      "talk": "https://en.wikipedia.org/wiki/Talk:Computer"
    },
    "mobile": {
      "page": "https://en.m.wikipedia.org/wiki/Computer",
      "revisions": "https://en.m.wikipedia.org/wiki/Special:History/Computer",
      "edit": "https://en.m.wikipedia.org/wiki/Computer?action=edit",
      "talk": "https://en.m.wikipedia.org/wiki/Talk:Computer"
    }
  },
  "api_urls": {
    "summary": "https://en.wikipedia.org/api/rest_v1/page/summary/Computer",
    "metadata": "https://en.wikipedia.org/api/rest_v1/page/metadata/Computer",
    "references": "https://en.wikipedia.org/api/rest_v1/page/references/Computer",
    "media": "https://en.wikipedia.org/api/rest_v1/page/media/Computer",
    "edit_html": "https://en.wikipedia.org/api/rest_v1/page/html/Computer",
    "talk_page_html": "https://en.wikipedia.org/api/rest_v1/page/html/Talk:Computer"
  },
  "extract": "A computer is a machine that can be instructed to carry out sequences of arithmetic or logical operations automatically via computer programming. Modern computers have the ability to follow generalized sets of operations, called programs. These programs enable computers to perform an extremely wide range of tasks. A \"complete\" computer including the hardware, the operating system, and peripheral equipment required and used for \"full\" operation can be referred to as a computer system. This term may as well be used for a group of computers that are connected and work together, in particular a computer network or computer cluster.",
  "extract_html": "<p>A <b>computer</b> is a machine that can be instructed to carry out sequences of arithmetic or logical operations automatically via computer programming. Modern computers have the ability to follow generalized sets of operations, called <i>programs.</i> These programs enable computers to perform an extremely wide range of tasks. A \"complete\" computer including the hardware, the operating system, and peripheral equipment required and used for \"full\" operation can be referred to as a <b>computer system</b>. This term may as well be used for a group of computers that are connected and work together, in particular a computer network or computer cluster.</p>"
}
```

### PR DESCRIPTION
**Pull request by marcosc90**
Fixes #6708

### CODE REVIEW

File: test_util/src/lib.rs [hunk 1]
```diff
@@ -155,6 +156,22 @@ pub async fn run_all_servers() {
   let absolute_redirect_server_fut =
     warp::serve(routes).bind(([127, 0, 0, 1], REDIRECT_ABSOLUTE_PORT));
 
+  let routes = warp::path!("a" / "b" / "c")
```
Comment (marcosc90) [inline test_util/src/lib.rs:159]: This is working fine, but I have less than 2 hours of rust coding in my life :stuck_out_tongue: , so feel free to improve it.

Comment (ry) [reply test_util/src/lib.rs:159]: can you say in words what the behavior of this is supposed to be? 

do none of the other servers cover this case?

(Just trying to push back on adding new servers here.)

Comment (marcosc90) [reply test_util/src/lib.rs:159]: I wanted to, but with my limited rust knowledge was the best I could do. Probably can be added very easily to one of the other servers. 

Need the request to redirect to the same server either with: `some-path` or `/some-path`, without causing a redirect loop.

Comment (marcosc90) [reply test_util/src/lib.rs]: @ry added the route to one of the other servers.

Comment (ry) [reply test_util/src/lib.rs]: nice!

**Pull request merged**